crates/bevy_render/src/extract_component.rs

```diff
@@ -205,33 +206,33 @@ impl<T: Asset> ExtractComponent for Handle<T> {
     }
 }
 
-/// This system extracts all components of the corresponding [`ExtractComponent`] type.
+/// This system extracts all synced entities components of the corresponding [`ExtractComponent`] type.
```

Comment:
This description does not make sense to me / is unclear. 

Correct me if I'm wrong, but this system/plugin does basically exactly what it did before this PR, _except_ the only new thing is that there's a new constraint: any entity with the component C, must also have SyncRenderWorld on it as well? @alice-i-cecile or ECS SMEs: Can we require that `C` here must also require SyncRenderWorld in the trait bounds? Is `C: Require<SyncRenderWorld>` a thing?

As for fixing the description, assuming I understand everything correctly, I would say "This system extracts all components of the corresponding [`ExtractComponent`] type, for entities that are synced via SyncRenderWorld", or something like that.